V1 Submission merge to master

DariaNico opened the pull request:
known bugs for v1:
- sometimes the light gets stuck and never gets smaller
- also on Daria's computer, the light decrease settings are really slow, but on Emily's they're just right

wigofifi committed:
Food interaction partially implemented

wigofifi committed:
Improved food interaction/despawn, decreased light radius, implemented slightly more variable food spawning

wigofifi committed:
Delete gameobject after hiding

wigofifi committed:
Use Emily's prefab for food

wigofifi committed:
Randomize spawn of food

wigofifi committed:
Refactored, removed unnecessary variables and functions that are declared but not used

wigofifi committed:
Fixed merge conflicts and improved collision size of food

wigofifi committed:
Player light and food light decrease over time. Currently no actual impact on gameplay.

DariaNico committed:
Merge pull request #54 from Hinute/feature/player-grows

Feature/player grows

DariaNico committed:
Merge pull request #56 from Hinute/feature/character-weakens-over-time

Feature/character weakens over time

DariaNico committed:
Reorganize prefab folder and move working food spawns and glowing player to game level 0

wigofifi committed:
Refining light growth and contraction

DariaNico committed:
Merge pull request #57 from Hinute/feature/merge-new-stuff-into-game-level-0

Reorganize prefab folder and move working food spawns and glowing pla…

DariaNico committed:
InitialSetup: move game level 0 to daria'ssandbox

wigofifi committed:
Changed food references to be held in a list so we can reference all food, not just most current

wigofifi committed:
Updated to level 1

Hinute committed:
Merge pull request #58 from Hinute/feature/speed-death-mechanics

Feature/speed death mechanics

DariaNico committed:
Add edge collider tailored to cave background

wigofifi committed:
Added pause menu

Hinute committed:
Merge pull request #59 from Hinute/feature/add-pause-menu

Added pause menu

wigofifi committed:
I don't know what this is, something that should've been committed onto develop already...

wigofifi committed:
Fixed issue with pause menu not pausing light dimming and food spawning

emberkheimer committed:
Logo and Select Sound

resized menu and put select sound in audio manager

DariaNico committed:
Make player bounce off cave walls

DariaNico committed:
Make CaveBackground with collider a prefab

DariaNico committed:
(m) reorganized GameLevel0Assets

wigofifi committed:
Player dies and max scores displays, added to Will's Sandbox

DariaNico committed:
Merge pull request #60 from Hinute/feature/logo-select

Logo and Select Sound

DariaNico committed:
Merge pull request #61 from Hinute/feature/add-game-boundaries-and-colliders

Feature/add game boundaries and colliders

wigofifi committed:
Probably added unnecessary things into Starts trying to fix issue that was caused by loading wrong scene

wigofifi committed:
Fixing rebase issue

wigofifi committed:
Switching to Daria's background and updating bright-level 0

Hinute committed:
Merge pull request #62 from Hinute/feature/player-death

Feature/player death

emberkheimer committed:
Wall sound and Rect Map

wigofifi committed:
Fixed issue where variable were not resetting after going back to menu after a death, leaving the death screen up when player attempts to play the next game.

DariaNico committed:
Merge pull request #63 from Hinute/feature/rectangle-map

Wall sound and Rect Map

DariaNico committed:
Merge pull request #64 from Hinute/bugfix/able-to-play-after-death

Fixed issue where variable were not resetting after going back to men…

DariaNico committed:
AudioManager Prefab & Bight-MainMenu Scene: Add all sounds and play correct sounds in main menu

DariaNico committed:
Bright-Level_0 & PlayerGlowy: Modify layer location and panel transparency

DariaNico committed:
Add sound to PauseMenu script

DariaNico committed:
Player Script: Refactor and add sounds to player

DariaNico committed:
Add death sound

DariaNico committed:
Bright-Level_0: Modify Death Menu

DariaNico committed:
Change main menu music and change button prefab select coloring

DariaNico committed:
Audio Manager: set initial music vol to .5

DariaNico committed:
Merge pull request #65 from Hinute/feature/attach-sound-to-game-events

Feature/attach sound to game events

DariaNico committed:
Bright-Level_0: Add collider to background. Also add safety net collider too

DariaNico committed:
Merge pull request #66 from Hinute/feature/add-background-collider-to-rect-background

Bright-Level_0: Add collider to background. Also add safety net colli…

DariaNico committed:
Bright-Level_0: make background into a Prefab

DariaNico committed:
Bright-Level_0: Add FoodSpawnGrounds to background and attach food controller there

DariaNico committed:
Make food spawn only within spawner

DariaNico committed:
New additions from a build

DariaNico committed:
Get menu button back on pause screen and update foodspawnground

DariaNico committed:
Get cave bump audio to work

DariaNico committed:
Merge pull request #67 from Hinute/feature/spawn-food-within-collider

Feature/spawn food within collider

DariaNico committed:
FoodController: Randomly generate colored foods

DariaNico committed:
FoodController: Change sprite color to match food color

DariaNico committed:
Daria's Sandbox and Player: Change player color on food consumption

Hinute commented:
PR for this is 2 weeks old, but pipeline is done. Does this need review? Does this need to be merged? @DariaNico

Hinute committed:
Merge pull request #69 from Hinute/feature/generate-colorful-foods

Feature/generate colorful foods

DariaNico commented:
Unsure..  @Hinute if we think it's jnbroken, we should def do a master merge though.

emberkheimer committed:
Color food sprites

sprites scaled and lit in prefab only-adding glow effect in next branch!

emberkheimer committed:
changed prefab folder

emberkheimer committed:
Merge pull request #87 from Hinute/feature/color-sprites

Feature/color sprites

wigofifi committed:
Added in game instruction menu

DariaNico committed:
MainMenu Scene: Make main menu show first. And reorder button order

DariaNico committed:
MainMenuScene: Make entirety of instructions menu show up properly on Mac

wigofifi committed:
Added missing 0 from light decrease speeds, presenting too quick on PCs. Fix ed issue where food doesn't continue spawning on timer after player dies once, becuase Player.isDead remains true by setting Player.isDead to true when user clicks on Play in Main Menu.

Hinute committed:
Merge pull request #89 from Hinute/feature/add-instruction-menu

Added in game instruction menu

Hinute committed:
Merge pull request #90 from Hinute/bugfix/radius-decrease-rate-and-food-spawn-after-dying

Added missing 0 from light decrease speeds, presenting too quick on P…

wigofifi committed:
Changed random.range to be 0,6 to hopefully include purple

wigofifi committed:
Changed main camera to black

emberkheimer committed:
all da food glows

the emission maps are fd up so ill fix these after doing more important stuff first

emberkheimer committed:
Accidently put an unrelated folder in Bright folder

emberkheimer committed:
Merge pull request #93 from Hinute/feature/food-glow

Feature/food glow

emberkheimer committed:
New idle anim player

swapped out sprite in level 0 glowy prefab and added animator

emberkheimer committed:
Merge pull request #94 from Hinute/feature/animated-player

New idle anim player

emberkheimer committed:
Better Player!

new shader- material on player sprite

emberkheimer committed:
Merge pull request #95 from Hinute/feature/player-shader

Better Player!

emberkheimer committed:
sound assets

switched gameplay music and added sound for damage

emberkheimer committed:
Merge pull request #96 from Hinute/feature/new-sounds

sound assets

wigofifi committed:
Switched to using new prefab food sprites. Fixed PlayerGlowy Collider being too large. Removed some code no longer necessary due to using new prefab food sprites (logic for setting random color for sprite and light no longer necessary, as this info is contained in the food object that we are choosing to spawn)

DariaNico committed:
Remove old unused prefabs and change up food, background, and player colliders

Hinute committed:
Merge pull request #98 from Hinute/feature/switch-to-new-food-sprites

Switched to using new prefab food sprites. Fixed PlayerGlowy Collider…

DariaNico committed:
Multi-food merged: change colliders to triggers

DariaNico committed:
MultiFood: Modify collider shapes to better fit the food

DariaNico committed:
FoodSpawnGrounds: Modify box demensions to better fit map

DariaNico committed:
Player + Wall collider: make the sound play when you hit a wall

DariaNico committed:
Merge pull request #102 from Hinute/bugfix/better-boundaries

Bugfix/better boundaries

wigofifi committed:
Refactored Update and Decrease Player Light to be more readable and do fewer things

Hinute committed:
Merge pull request #103 from Hinute/bugfix/refactoring-player-class

Refactored Update and Decrease Player Light to be more readable and d…

DariaNico committed:
Modify readme

DariaNico committed:
Add to instructions on the main menu

DariaNico committed:
Death Screen: add replay button

DariaNico committed:
PausePanel: Add additional text to explain how to resume the game

DariaNico committed:
AudioManager: Add audio easter egg to old music if you take a long time

DariaNico committed:
Stop level music if returning to main menu

DariaNico committed:
Modify gitignore

emberkheimer committed:
astar

emberkheimer committed:
Astar package

wigofifi committed:
Added missing files

emberkheimer committed:
Working Enemy

enemy in scene "Em2"

wigofifi committed:
Manually set framerate and improve the speed of light change to match new framerate.

emberkheimer committed:
Particle Shoot

spacebar shoots particles from player

emberkheimer committed:
Particles kill enemy!

shoot and enemy damage scripts

emberkheimer committed:
Red Eyes

what i have rn and also now what the enemies have

wigofifi committed:
Added damage to player if player collides with enemy

Hinute committed:
Merge pull request #104 from Hinute/bugfix/set-framerate-and-improve-speeds

Manually set framerate and improve the speed of light change to match…

wigofifi committed:
Added ability for player to win upon reaching a light radius of 3, with code in place to start next level

wigofifi committed:
Switched new scene load to happen last

Hinute committed:
Merge pull request #105 from Hinute/feature/reimplement-enemies

Feature/reimplement enemies

wigofifi committed:
Fix detection of enemy collision

Hinute committed:
Merge pull request #106 from Hinute/feature/add-win-condition

Feature/add win condition

wigofifi committed:
Added references into Scene Em2 so that we can move player in that level

DariaNico committed:
Fix Instructions, food spawn grounds, and main menu audio manager

DariaNico committed:
Last minute bugfixes to make level progression work

DariaNico committed:
(optional) for 12/13/2020 submission: change values to have the game be winable
